src/NetTopologySuite.IO.ShapeFile/Shapefile.cs

```diff
@@ -20,15 +20,23 @@ public partial class Shapefile
         /// <returns>The equivalent for the geometry object.</returns>
         public static ShapeGeometryType GetShapeType(Geometry geom)
         {
-            geom = GetNonEmptyGeometry(geom);
+            var fixedGeom = GetNonEmptyGeometry(geom);
```

Comment:
Maybe we should rename `GetNonEmptyGeometry` so it's clearer that it will return:

- `null` if the input is `null`, empty, or a `GeometryCollection` that only contains `null` or empty things
- the input `Geometry`, if it's not a `GeometryCollection`
- the first non-empty **direct** child of the `GeometryCollection`

As it's named (and used) right now, `GetNonEmptyGeometry` sounds like it would return the input if it's non-empty, or `null` otherwise...

Reply:
why this check is useful? I think the right thing to do is:

- return `null` if geom is null 
- return `null` if geom is an empty geom
- return `null` if geom is a collection with empty childs
- return the `input geom` if not a collection
- return the `input geom` is is a collection

all three null/empty case are done via a simple 
```
if (geom == null || geom.IsEmpty)
                return null;
```

Reply:
not actually true, collections are not empty is at most a geometry is not empty

Reply:
> why this check is useful?

I don't think it is, but that's what it's doing right now, so this suggestion was to make it clearer in the caller's code **why** we need to adjust the return value even if we give it a `MultiPoint`